src/appleseed/foundation/meta/tests/test_imagefilewriter.cpp

```diff
@@ -0,0 +1,245 @@
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.bmp";
+        
+        {
+            Image image(2, 2, 2, 2, 3, PixelFormatUInt16);
+            
+            image.clear(Color3b(50, 100, 150));
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color3b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(Color3b(50, 100, 150), c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteEXR_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.exr";
+        
+        {
+            Image image(2, 2, 2, 2, 4, PixelFormatUInt16);
+            
+            image.clear(Color4b(50, 100, 150, 42));
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color4b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(Color4b(50, 100, 150, 42), c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteHDR_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.hdr";
+        
+        {
+            Image image(2, 2, 2, 2, 3, PixelFormatUInt16);
+            
+            image.clear(Color3b(50, 100, 150));
+            
+            GenericImageFileWriter writer(ImageFilePath);
+            writer.append_image(&image);
+            writer.write();
+        }
+        
+        {
+            GenericImageFileReader reader;
+            std::unique_ptr<Image> image(reader.read(ImageFilePath));
+
+            for (size_t y = 0; y < 2; ++y)
+            {
+                for (size_t x = 0; x < 2; ++x)
+                {
+                    Color3b c;
+                    image->get_pixel(x, y, c);
+                    EXPECT_EQ(Color3b(50, 100, 150), c);
+                }
+            }
+        }
+    }
+    
+    TEST_CASE(WriteJPG_CorrectlyWritesImagePixels)
+    {
+        const char* ImageFilePath = "unit tests/outputs/test_imagefilewriter.jpg";
+        
+        {
+            Image image(2, 2, 2, 2, 3, PixelFormatFloat);
+            
+            image.clear(Color3f(0.6f, 0.7f, 0.8f));
```

Comment:
I would prefer that we stick to integer components, especially for JPEG which doesn't support floating point components. You can use `std::abs()` in conjunction with `EXPECT_TRUE()` to test that values that you read back are close enough to input values.